langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibDecimalTest.java

```diff
@@ -216,6 +216,60 @@ public void testRoundAsMethodInvok(BDecimal arg, String expected) {
         };
     }
 
+    @Test(dataProvider = "decimalRoundToFractionDigitsValueProvider")
+    public void testRoundToFraction(BDecimal arg, Integer fractionDigits, String expected) {
+        Object returns = BRunUtil.invoke(compileResult, "testRoundToFractionDigits",
+                new Object[] {arg, fractionDigits});
+        assertEquals(returns, ValueCreator.createDecimalValue(expected));
+    }
+
+    @DataProvider(name = "decimalRoundToFractionDigitsValueProvider")
+    public static Object[][] decimalRoundToFractionDigitsValueProvider() {
+        return new Object[][] {
+                {ValueCreator.createDecimalValue("-504023030303030303030.303567"), 3, "-504023030303030303030.304"},
+                {ValueCreator.createDecimalValue("-50402.303567"), 8, "-50402.303567"},
+                {ValueCreator.createDecimalValue("3.303567"), 0, "3"},
+                {ValueCreator.createDecimalValue("3234.5"), 0, "3234"},
+                {ValueCreator.createDecimalValue("3234.303567"), 0, "3234"},
+                {ValueCreator.createDecimalValue("3234.503567"), 0, "3235"}, // not sure
+                {ValueCreator.createDecimalValue("3233.503567"), 0, "3234"},
+                {ValueCreator.createDecimalValue("-3234.303567"), 0, "-3234"},
+                {ValueCreator.createDecimalValue("-3234.5"), 0, "-3234"},
+                {ValueCreator.createDecimalValue("3.303567"), -1, "0"},
+                {ValueCreator.createDecimalValue("3234.303567"), -2, "3.2E+3"},
+                {ValueCreator.createDecimalValue("3234.303567"), -10, "0"},
+                {ValueCreator.createDecimalValue("3234.303567"), 7, "3234.303567"},
+                {ValueCreator.createDecimalValue("3234.303567"), 10, "3234.303567"},
+                {ValueCreator.createDecimalValue("5.5555"), 4, "5.5555"},
+                {ValueCreator.createDecimalValue("5.5555"), 3, "5.556"},
+                {ValueCreator.createDecimalValue("5.5555"), 2, "5.56"},
+                {ValueCreator.createDecimalValue("5.56"), 1, "5.6"},
+                {ValueCreator.createDecimalValue("5.55"), -1, "1E+1"}
+        };
+    }
```

Comment:
Shall we add some tests when the x is close and x is equal to java Integer.MAX_VALUE/Integer.MIN_VALUE
just to check branches available in round function?

Reply:
Thanks added by 2d77547637463307e7bfd4333e4a6d087114b7fd